lib/src/configuration.rs

```diff
@@ -0,0 +1,59 @@
+#[derive(Debug, Clone)]
+pub struct Configuration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub basic_auth: Option<BasicAuth>,
+    pub oauth_access_token: Option<String>,
+    pub bearer_access_token: Option<String>,
+    pub api_key: Option<ApiKey>,
```

Comment:
Do we really want all of these to be public?

Reply:
@uniqueg, I had a talk with @pavelnikonorov, and I think for now, we can keep the parameters public as it isn't a security risk for now, but later on we could use getter and setters